### PR TITLE
[ONNX] Fix view for dynamic input shape

### PR DESCRIPTION
**Pull request by neginraoof**
Export of view op with dynamic input shape is broken when using tensors with a 0-dim.
This fix removes symbolic use of static input size to fix this issue.

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -2427,6 +2427,19 @@ def forward(self, input, other):
...
+    def test_view_dynamic_zero_dim(self):
+        class ViewModel(torch.nn.Module):
+            def forward(self, input):
+                input = input.view(-1, 2)
+                return input.view(1, -1)
+
+        x = torch.ones(2)
+        another_x = torch.empty((0,))
+        model = ViewModel()
+        print(model(x).shape)
```
Comment (dashesy) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:2439]: does this print affect the test?

**Comment by BowenBao** [comment]
failed tests may be related
```
Aug 27 19:29:00 FAILED test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset7::test_view_dynamic_zero_dim
Aug 27 19:29:00 FAILED test/onnx/test_pytorch_onnx_onnxruntime.py::TestONNXRuntime_opset8::test_view_dynamic_zero_dim
```

**Comment by neginraoof** [comment]
cc @bzinodev Jenkins failures are unrelated. Can we re-trigger these to unblock merge?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -462,13 +462,10 @@ def view(g, self, size):
     if sym_help._is_value(size):
         shape = size
     else:
-        if self.isCompleteTensor():
-            self_sizes = self.type().sizes()
-            if self_sizes and len(size) == 2 and self_sizes[0] == size[0]:
-                return g.op("Flatten", self, axis_i=1)
         shape = g.op("Constant", value_t=torch.LongTensor(size))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:465]: I'm wondering if we need this step anymore. This seems due to that we are parsing `size` to get const by `is`, and now putting it back to tensor through `Constant` node. Maybe we can remove all these, and just directly return `g.op("Reshape", self, size)`

**Comment by BowenBao** [comment]
Test failures are related, seems related to my last suggestion.

**Comment by BowenBao** [comment]
test failures seem related

**Comment by neginraoof** [comment]
@BowenBao 
Thanks! I updated the symbolic. Looks like the line ```size = sym_help._maybe_get_const(size, 'is')``` in view symbolic is required to handle cases where the view size is an empty list.

**Review by BowenBao — APPROVED** [review]
LGTM, thanks!

**Review by bzinodev — APPROVED** [review]
